libyul/optimiser/ExpressionSplitter.cpp

```diff
@@ -68,7 +68,7 @@ void ExpressionSplitter::operator()(Switch& _switch)
 
 void ExpressionSplitter::operator()(ForLoop& _loop)
 {
-	(*this)(_loop.pre);
+	assertThrow(_loop.pre.statements.empty(), OptimizerException, "");
```

Comment:
I don't see a reason why the ExpressionSplitter should require empty pre blocks (please also change the header file and remove the requirement).